secret_store/src/key_server_cluster/decryption_session.rs

```diff
@@ -102,8 +102,10 @@ pub enum SessionState {
 
 impl Session {
 	/// Create new decryption session.
-	pub fn new(id: SessionId, access_key: Secret, self_node_id: Public, encrypted_data: EncryptedData, acl_storage: Arc<AclStorage>, cluster: Arc<Cluster>) -> Self {
-		Session {
+	pub fn new(id: SessionId, access_key: Secret, self_node_id: Public, encrypted_data: EncryptedData, acl_storage: Arc<AclStorage>, cluster: Arc<Cluster>) -> Result<Self, Error> {
```

Comment:
this function is getting quite long...maybe extract out to a `Params` struct for readability in invocations?